Ehr_OOP.py

```diff
@@ -0,0 +1,211 @@
+"""Bio stats assignment #OOP Mohammad Anas."""
+
+
+from datetime import datetime
+from typing import Union
+
+
+class Patient:
+    """Store a patient data."""
+
+    def __init__(
+        self,
+        PatientID: str,
+        gender: str,
+        DOB: Union[str, datetime],
+        race: str,
+        labs: list[str],
+    ) -> None:
+        """Instantiate Patient Class."""
+        self.patientID: str = PatientID  # O(1)
+        self.gender: str = gender  # O(1)
+        self.DOB: datetime = datetime.strptime(DOB, "%Y-%m-%d %H:%M:%S.%f")  # O(2)
+        self.race: str = race  # O(1)
+        self.labs: list[str] = labs  # O(1)
+        # class instantiation takes O(6) complexity
+
+    def __str__(self):
+        """Patient Class Representation."""
+        return self.patientID + " Class"  # O(1)
+
+    @property
+    def age(self):
+        """Calculate patient age."""
+        today = datetime.today()  # O(2)
+        return int((today - self.DOB).days / 365.25)  # O(4)
+
+    @property
+    def age_first_admission(self):
+        """Calculate age at first lab test."""
+        # get min lab date for the patient
+        mindate = min([i.labDate for i in self.labs])  # O(P)
+        # get age at the first lab visit
+        return int((mindate - self.DOB).days / 365.25)  # O(4)
+
+
+class Lab:
+    """Store Lab data."""
+
+    def __init__(
+        self,
+        Labname: str,
+        units: str,
+        value: Union[str, float],
+        labDate: Union[str, datetime],
+    ) -> None:
+        """Instantiate Lab Class."""
+        self.labname: str = Labname  # O(1)
```

Comment:
```suggestion
        self.name: str = name  # O(1)
```
We already know that it's a lab...